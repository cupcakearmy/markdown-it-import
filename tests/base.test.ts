import { describe, expect, test } from 'bun:test'
import { render } from './utils.ts'

describe('base', () => {
  test('no imports', async () => {
    expect(await render('simple.md')).toMatchSnapshot()
  })

  test('single import', async () => {
    expect(await render('whole.md')).toMatchSnapshot()
  })

  test('multiple imports', async () => {
    expect(await render('multiple.md')).toMatchSnapshot()
  })

  test('fail on not found', async () => {
    expect(() => render('notFound.md')).toThrow(/cannot locate file.*nirvana\.md/)
  })

  test('custom lines', async () => {
    expect(await render('partial.md')).toMatchSnapshot()
  })
})
