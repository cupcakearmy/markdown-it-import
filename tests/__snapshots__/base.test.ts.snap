// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`base no imports 1`] = `
"<h1>Some title</h1>
<ul>
<li>Some</li>
<li>List</li>
<li>Three</li>
</ul>
<blockquote>
<p>Some note</p>
</blockquote>
<table>
<thead>
<tr>
<th>Column A</th>
<th>Column B</th>
<th>Column C</th>
</tr>
</thead>
<tbody>
<tr>
<td>Some</td>
<td><code>stuff</code></td>
<td></td>
</tr>
</tbody>
</table>
"
`;

exports[`base single import 1`] = `
"<p>Here is a typescript snippet</p>
<pre><code class="language-ts">export function sum(a: number, b: number): number {
  return a + b
}

</code></pre>
"
`;

exports[`base multiple imports 1`] = `
"<p>Here is a typescript snippet</p>
<pre><code class="language-ts">export function sum(a: number, b: number): number {
  return a + b
}

</code></pre>
<blockquote>
<p>This is some amazing quote</p>
</blockquote>
"
`;

exports[`base custom lines 1`] = `
"<h1>Bubble sort in go</h1>
<pre><code class="language-go">func bubbleSort(arr []int) {
	len := len(arr)
	for i := 0; i &lt; len-1; i++ {
		for j := 0; j &lt; len-i-1; j++ {
			if arr[j] &gt; arr[j+1] {
				arr[j], arr[j+1] = arr[j+1], arr[j]
			}
		}
	}
	fmt.Println(&quot;\nAfter Bubble Sorting&quot;)
	for _, val := range arr {
		fmt.Println(val)
	}
}
</code></pre>
"
`;
